packages/good-design/src/core/animated/AnimatedProgress.tsx

```diff
@@ -0,0 +1,76 @@
+import React, { useEffect, useRef } from "react";
+import { Animated, Platform } from "react-native";
+import { View } from "native-base";
+
+import { withTheme } from "../../theme/hoc/withTheme";
+
+interface ITestAnimatedProps {
+  containerStyles?: object;
+  progressStyles?: object;
+  progressBar?: object;
+  animatedValue: number;
+  startValue: number;
+}
+
+export const theme = {
+  baseStyle: {
+    containerStyles: {
+      flex: 1,
+      alignItems: "center",
+      justifyContent: "center",
+      width: "100%"
+      // backgroundColor: "black"
+    },
+    progressStyles: {
+      width: "80%",
+      height: 1,
+      backgroundColor: "goodGrey.300",
+      borderRadius: 5,
+      overflow: "hidden"
+    },
+    progressBar: {
+      height: "100%",
+      backgroundColor: "#00AFFF"
+    }
+  }
+};
+
+// based on 3 steps progress bar
+const AnimatedProgress = withTheme({ name: "AnimatedProgress" })(
+  ({ containerStyles, progressStyles, progressBar, animatedValue, startValue, ...props }: ITestAnimatedProps) => {
```

Comment:
you need just **one** value,

- animatedValue is some animation ref defined inside 
- no startValue, you start from the **previous value** of value 
- if current > previous, you **run** animation
- otherwise no

Reply:
changed now to only startValue. did need a resetValue as well

Reply:
updated now to make the animated progress more abstract. just relies on a start and end value